soc/arm/Kconfig

```diff
@@ -34,7 +34,6 @@ config CPU_HAS_NRF_IDAU
 
 config CPU_HAS_FPU_DOUBLE_PRECISION
 	bool
-	depends on CPU_CORTEX_M7
```

Comment:
I think this should still depend on the arch variants, no?

Reply:
Why?

Reply:
> I think this should still depend on the arch variants, no?

@ioannisg Technically, that is right if we want to double-check arch support, but this list is only going to grow as more ARM variants are added. Also as @SebastianBoe pointed out, removing this line will allow the use of out-of-tree CPUs.

Reply:
For instance, we do not double-check if a CPU supports MPU or not using `depends on`.

https://github.com/zephyrproject-rtos/zephyr/blob/c7044958073e8c41299e7f829d97071201827cec/soc/arm/Kconfig#L6-L11

Reply:
This double-checking is not worth the extra maintenance and out-of-tree issues IMHO.

Reply:
All CPUs support the ARM MPU (except cortex-m0, but m0+ does)
Double-precision FPU is Cortex-M7 only, so I was thinking that's a protection against accidental enabling

Reply:
It's no longer M7 only, so this protection is causing more trouble than it's worth.

Reply:
..yeah, from Cortex-m side, it's M7, but sure i am ok if we wanna get rid of it
Checking with @galak as well :)

Reply:
Just to elaborate, it might become something like in the future:
`depends on (CPU_CORTEX_M7 || CPU_CORTEX_R4 || CPU_CORTEX_R5 || CPU_CORTEX_R7 || CPU_CORTEX_R8 || CPU_CORTEX_R52 || CPU_CORTEX_A8 || CPU_CORTEX_A9 || CPU_CORTEX_A15 || CPU_CORTEX_A53 || ...)`

Reply:
Well, if could be CPU_CORTEX_R  || CPU_CORTEX_A, exactly as we have CPU_CORTEX_M.
But, that's fine, I am convinced :) Waiting for @galak to comment :p